src/ui/public/filter_bar/lib/disable_filter.js

```diff
@@ -0,0 +1,25 @@
+export function disableFilter(filter) {
+  return setFilterDisabled(filter, true);
+}
+
+export function enableFilter(filter) {
+  return setFilterDisabled(filter, false);
+}
+
+export function setFilterDisabled(filter, disabled) {
+  const { meta = {} } = filter;
+
+  return {
```

Comment:
I have found that the best way to use javascript objects immutably is to return new objects when they are modified and original objects when they are not. This has the effect, that cheap comparisons by identity work well to detect changes while still keeping memory overhead at bay. It almost resembles structural sharing as used by the popular persistent data structure implementations and is in line with the best practice used by the redux and react communities.

The obvious downside is that it does not protect you against "malicious" code that modifies deep subtrees in-place, but I hope we will rather change that code than resolve to overly defensive programming.